proposals/0000-split-jest-from-react-native-core.md

```diff
@@ -0,0 +1,106 @@
+- **Reduced Maintenance Burden:** By making Jest optional, the React Native core team can focus on the maintenance of React Native itself and signals that maintenance of the Jest preset isn't coupled to React Native releases necessarily. This is in line with the Learn Core initiative.
+
+In short, this change affirms Jest being a choice and comes with maintenance and installation speed benefits.
+
+## Detailed design
+
+1. Create a separate package, `@react-native/jest-preset`
+2. The current `jest` sub-folder and `jest-preset.js` move to the new package
+3. `@jest/create-cache-key-function`, `babel-jest`, `jest-environment-node`, `babel-plugin-syntax-hermes-parser`
+4. Update template projects, as appropriate to switch from `react-native/jest-preset` to `@react-native/jest-preset`
+5. Update documentation, as needed to switch from `react-native/jest-preset` to `@react-native/jest-preset`
+
+**Note:** There's an implicit usage of `react-test-renderer` in the `react-native/jest` folder. This package is deprecated, and moving this out of `react-native` also removes the deprecated testing recommendation from `react-native`.
+
+### Migration Path
+
+- Install `@react-native/jest-preset` as a dev-dependency
+- Switch `preset` in your Jest configuration to `@react-native/jest-preset`
+
+## Drawbacks
+
+- **Migration**: Existing projects will need to add the new dependency
+- **Documentation:** If any documentation assumed Jest, it will now need to point at new instructions or clarify that Jest is optional.
+- **Fragmentation:** While Jest may not be the best choice for every project, this removes a default choice, which may mean beginners are faced with an additional decision.
+- **Template**: Templates will have to choose whether or not to include Jest and the preset.
+
+## Alternatives
+
+- **Continue including Jest:** We could continue including the preset, and take on the maintenance burden associated with this, as long as development and support is revilatised.
+- **Switch default testing framework:** We could replace Jest with another test framework and runner like Vitest. However, this increases maintenance burden.
+- **Minimal testing interface:** We could create a framework-agnostic unit testing interface. However, `@testing-library/react-native` already exists.
+
+This proposal was chosen partially because options for E2E testing already exist, and alternative runners can use `@testing-library/react-native`. The main complexity of replacing Jest for React Native is associated with bundling. However, as new projects make strides in supporting more projects, as long as native-platform-extension support and Babel preset transpilation of all modules is possible, a test framework can start supporting React Native. Removing the `jest-preset` from `react-native` signals to contributors to encourage other test runners to support React Native, and encourages external maintenance of new test runners.
```

Comment:
FWIW, `@testing-library/*` also runs in other systems. Like:
- Bun: https://bun.com/guides/test/testing-library
- Vitest: https://testing-library.com/docs/svelte-testing-library/setup/